tools/src/wyvern/target/corewyvernIL/transformers/ILTransformer.java

```diff
@@ -0,0 +1,9 @@
+package wyvern.target.corewyvernIL.transformers;
+
+import wyvern.target.corewyvernIL.expression.IExpr;
+
+public interface ILTransformer {
+
+	public IExpr transform(IExpr ast);
```

Comment:
This approach ends up requiring a lot of instanceof tests in the implementation of transform.  Could this be implemented as a wyvern.target.corewyvernIL.astvisitor.ASTVisitor?

Note that ASTVisitor has an unfortunate concrete dependence on an "OIREnvironment."  This dependence should be removed, but in the meantime you could use it and just pass in null there.